tests/functional/test_integration.py

```diff
@@ -43,27 +67,39 @@ async def test_build_and_deploy_with_psql(
         check=True,
     )
     await ops_test.model.wait_for_idle(status=ActiveStatus.name, raise_on_error=False)
-    subprocess.check_output(
-        ["juju", "destroy-model", influx_model_name, "--force", "--destroy-storage", "-y"]
+    await ops_test.juju(
+        "destroy-model",
+        f"{controller_name}:{influx_model_name}",
+        "--force",
+        "--destroy-storage",
+        "-y",
     )
-    subprocess.check_output(["juju", "switch", "k8s-ctrl"])
 
 
 async def test_status(ops_test: OpsTest):
+    """
+    arrange: given the resulting juju model of the first test
+    assert: the model should have all its charms in active state.
+    """
     assert ops_test.model.applications["gunicorn-k8s"].status == ActiveStatus.name
     assert ops_test.model.applications["postgresql-k8s"].status == ActiveStatus.name
 
 
 async def test_workload_psql_var(ops_test: OpsTest):
+    """
+    arrange: given the resulting juju model of the first test
```

Comment:
Does this test depend on being executed after the other tests? Or does it just depend on all the applications being deployed?

Reply:
It might be worth clarifying this now since the change to the fixture I'm guessing this doesn't depend on the first test anymore